### PR TITLE
[8.0] Add module to manage GLN partner identification number

### PR DESCRIPTION
**Pull request by rousseldenis**
Depends on [PR #296 ](https://github.com/OCA/partner-contact/pull/296)
Update : Rebased (PR 296 was merged)

### CODE REVIEW

**Comment by rousseldenis** [comment]
@lmignon @andhit-r Don't hesitate to review !
@pedrobaeza Can you add label need review ?

**Review by andhit-r — CHANGES_REQUESTED** [review]
@rousseldenis thank you for your contributions. You might wanna check your commit history. There are a lot of file changes that is not supposed to happend.

**Comment by rousseldenis** [comment]
@andhit-r Solved!

File: partner_identification_gln/README.rst
Comment (andhit-r) [inline partner_identification_gln/README.rst]: Wrong runbot link

File: partner_identification_gln/__openerp__.py
Comment (andhit-r) [inline partner_identification_gln/__openerp__.py]: Remove empty key

File: partner_identification_gln/static/description/icon.svg
Comment (andhit-r) [inline partner_identification_gln/static/description/icon.svg:1]: Why do you use .svg not .png file?

**Review by andhit-r — CHANGES_REQUESTED** [review]
Functional: No test
Code: minor comments inline

File: partner_identification_gln/static/description/icon.svg
Comment (rousseldenis) [reply partner_identification_gln/static/description/icon.svg:1]: @andhit-r Both apparently accepted as in module template https://github.com/OCA/maintainer-tools/tree/master/template/module/static/description

**Comment by rousseldenis** [comment]
@andhit-r Changes done!

**Review by andhit-r — APPROVED** [review]
:+1:

**Comment by rousseldenis** [comment]
@lmignon A little thumb up ? ;-)

**Comment by andhit-r** [comment]
@mikevhe18 @azmimr67 please help review. Thanks

File: partner_identification_gln/models/partner.py
Comment (lmignon) [inline partner_identification_gln/models/partner.py]: Active test is by default into domain. Am I wrong?

File: partner_identification_gln/models/partner.py
Comment (lmignon) [inline partner_identification_gln/models/partner.py]: The search for duplicate should be put in a dedicated method to avoid code duplication with line 32

**Review by lmignon — CHANGES_REQUESTED** [review]

File: partner_identification_gln/models/partner.py
Comment (rousseldenis) [reply partner_identification_gln/models/partner.py]: We are not on the base model but on comodel (we don't search on res.partner).
If a partner is inactive, the search will return its numbers. We don't want this behaviour.

**Comment by rousseldenis** [comment]
@lmignon that's because the context is not exactly the same.

In one case we have an EAN, in the other not.
And the cases concern different categories of id_numbers (we can have one of each in one partner).

I think the most disturbing thing is that I use the same variable name duplicate_gln. I should rename one in duplicate_gcp

**Comment by azmimr67** [comment]
👍testes

**Comment by rafaelbn** [comment]
Hi @lmignon , could you please unblock this PR? thanks! 😄

**Review by lmignon — APPROVED** [review]
@rafaelbn OK for me :smirk:

**Comment by pedrobaeza** [comment]
Please squash at your taste

**Comment by rafaelbn** [comment]
Could I squash ?

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

**Comment by rafaelbn** [comment]
Hi @rousseldenis are you going to squash please? thanks

**Comment by lmignon** [comment]
@rafaelbn @rousseldenis I'll squash

**Comment by lmignon** [comment]
@rafaelbn Squash done.

**Pull request merged**
